swap/src/api/request.rs

```diff
@@ -0,0 +1,861 @@
+                             let our_peer_id = swarm.local_peer_id();
+
+                             tracing::debug!(peer_id = %our_peer_id, "Network layer initialized");
+
+                             for seller_address in seller_addresses {
+                                 swarm
+                                     .behaviour_mut()
+                                     .add_address(seller_peer_id, seller_address);
+                             }
+
+                             let (event_loop, event_loop_handle) =
+                                 EventLoop::new(swap_id, swarm, seller_peer_id)?;
+                             let handle = tokio::spawn(event_loop.run().instrument(Span::none()));
+
+                             let monero_receive_address = context.db.get_monero_address(swap_id).await?;
+                             let swap = Swap::from_db(
+                                 Arc::clone(&context.db),
+                                 swap_id,
+                                 Arc::clone(
+                                     context
+                                         .bitcoin_wallet
+                                         .as_ref()
+                                         .context("Could not get Bitcoin wallet")?,
+                                 ),
+                                 Arc::clone(
+                                     context
+                                         .monero_wallet
+                                         .as_ref()
+                                         .context("Could not get Monero wallet")?,
+                                 ),
+                                 context.config.env_config,
+                                 event_loop_handle,
+                                 monero_receive_address,
+                             )
+                                 .await?;
+
+                             tokio::select! {
+                                 event_loop_result = handle => {
+                                     event_loop_result?;
+                                 },
+                                 swap_result = bob::run(swap) => {
+                                     swap_result?;
+                                 }
+                             };
+                             Ok::<(), anyhow::Error>(())
+                        } => {
+                            ()
+                        },
+                        _ = context.swap_lock.listen_for_swap_force_suspension() => {
+                             tracing::debug!("Shutdown signal received, exiting");
+                             ()
+                         }
+                    }
+                    context
+                        .swap_lock
+                        .release_swap_lock()
+                        .await
+                        .expect("Could not release swap lock");
+                }.instrument(Span::none()));
+                Ok(json!({
+                    "result": "ok",
```

Comment:
We always return "ok" here even if the swap does not exist or something goes wrong. I think its fine as is but if someone knows how to implement this such that we fail if we encounter an error in the very beginning, we could do that